extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java

```diff
@@ -614,6 +614,94 @@ private Table maybeFixBigDecimal(Table toFix) {
+        }
+        final ParquetMetadata metadata = new ParquetTableLocationKey(new File(path), 0, null).getMetadata();
+        assertTrue(metadata.getFileMetaData().getKeyValueMetaData().get("deephaven").contains("\"version\":\"0.4.0\""));
+
+        path = ParquetTableReadWriteTest.class.getResource("/ReferenceParquetVectorData.parquet").getFile();
+        ParquetTools.readTable(new File(path)).select();
+
+        path = ParquetTableReadWriteTest.class.getResource("/ReferenceParquetArrayData.parquet").getFile();
+        ParquetTools.readTable(new File(path)).select();
+    }
+
+    @Test
+    public void testVersionChecks() {
+        assertFalse(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.0.0"));
+        assertFalse(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.4.0"));
+        try {
+            ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.3");
+            TestCase.fail("Exception expected for invalid version string");
+        } catch (IllegalArgumentException expected) {
+        }
+        assertTrue(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.31.0"));
+        assertTrue(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.31.1"));
+        assertTrue(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("0.32.0"));
+        assertTrue(ColumnChunkPageStore.satisfiesMinimumVersionRequirements("1.3.0"));
+    }
+
+    @Test
+    public void testWritingDifferentPageSizes() {
+        // Make a table with arrays of decreasing sizes such that different pages will have different number of rows
+        Table arrayTable = TableTools.emptyTable(100).update(
+                "intArrays = java.util.stream.IntStream.range(0, i).toArray()").reverse();
+        final File dest = new File(rootFile + File.separator + "testWritingDifferentPageSizes.parquet");
+        final ParquetInstructions writeInstructions = new ParquetInstructions.Builder()
+                .setTargetPageSize(ParquetInstructions.MIN_TARGET_PAGE_SIZE)
+                .build();
+        ParquetTools.writeTable(arrayTable, dest, writeInstructions);
+        Table fromDisk = ParquetTools.readTable(dest).select();
+        TstUtils.assertTableEquals(arrayTable, fromDisk);
+
+        // Make a table such that only the last page has different number of rows, all else have equal number
+        final long NUM_ROWS = 1000;
+        arrayTable = TableTools.emptyTable(NUM_ROWS).update(
+                "intArrays = (i <= 900) ? java.util.stream.IntStream.range(i, i+50).toArray() : " +
+                        "java.util.stream.IntStream.range(i, i+2).toArray()");
+        ParquetTools.writeTable(arrayTable, dest, writeInstructions);
+        fromDisk = ParquetTools.readTable(dest);
+        // Access something on the last page to make sure we can read it
+        final int[] data = (int[]) fromDisk.getColumnSource("intArrays").get(998);
+        assertTrue(data.length == 2 && data[0] == 998 && data[1] == 999);
```

Comment:
I'm not a fan of chaining multiple asserts together w/ an `&&` like this; each should be tested w/ more specific method:

```java
assertEquals(2, data.length);
assertEquals(998, data[0]);
...
```